### DISCUSSION TITLE
[Feature Request] monaco-editor json language how to add variable?

### DISCUSSION THREAD

**Discussion by sunyuanonly** [microsoft/monaco-editor#4116] [Q&A]
### Context

- [X] This issue is not a bug report. *(please use a different template for reporting a bug)*
- [X] This issue is not a duplicate of an existing issue. *(please use the [search](https://github.com/microsoft/monaco-editor/issues) to find existing issues)*

### Description

 json language how to add variable and skip verification

### Monaco Editor Playground Link

Postman
![image](https://github.com/microsoft/monaco-editor/assets/16498201/20885542-ba40-4298-ae01-b203d9a32b67)

monaco-editor
![1691123549490](https://github.com/microsoft/monaco-editor/assets/16498201/e23772e8-b4d3-418f-ad22-569fb1ed40d7)

**Comment by jhegele** [comment]
Were you ever able to solve this? I'm trying to implement very similar functionality and haven't been able to figure it out.

**Comment by MolikoDeveloper** [comment]
after a quick search, im gonna implement a new language to my project..

json is implemented as native in monaco, becuase is a simple language, it does not requires all the features monaco implement.

but, i need it for exactly the same reason you need.
for now, this is my [playground](https://microsoft.github.io/monaco-editor/playground.html?source=v0.52.2#XQAAAAJmCAAAAAAAAABBqQkHQ5NjdMjwa-jY7SIQ9S7DNlzs5W-mwj0fe1ZCDRFc9ws9XQE0SJE1jc2VKxhaLFIw9vEWSxW3yscxEKMmu_1uhkuiAXzpN6YRsnxkPkuTPpX83j-vV0qCn8xx14YIsCYjZ6tx2WDkZ3vUvzdAvW2kKdfZSP547Sshkkb6Mo0lQw1rSkI5pm4TFASCKh13Sy72l7w5_uY2xR7Gh4HvZt4rsgfgd4XSTmqTxvKQBjphtPSWvgn0Q1iEPGh_lpyLccKiidi_U1-5bgmKZg5iCmtC0RG2rLSBiPne5agpLWya4fFYi75dXnDa1iGzAYshbk59opQjMPWNZU6s3wXx4iTmLogqgWpNg_ZERTtNyekU5EMlmXUDpHmCWa4CCAqS82lJyScttLPtqPhJv9pPhsZcKi6g8bxMDjJ0a6UdUtt3-wROdznii6Q4WVz1Nznpj4eo_aFsw6CgAYMURFC0evXscdhpP1HvRwOrIqxCzH_I0IxdvYUmsUmDVa8G1i6_HX5TPaBRvE1_waHTQHAfuvRgbBqV7hhd1lOjEypCB2_5ihMdcjVf650JXrRHQh9B05C3bN4zxr06J0JRtFl7nHGxSgfZiBbllWbvxGrpoWjErwXRrd4ySA4F4_4kEQf1BURnt9ad29AWdK6wJ7hbQk0vTENuucPIqfy7HKqqLA5qnPMOX6IZ_WPp0CRNR3JNJeL1FvYYLKzVKbH5bCk9sZuR1XEIZpgw-E1K-OGf53ns9LkGE7CVoNKSlvg1ANzZApjb0mOe9ieMu_vtuRyv51ERk2QDpqtPArnFIhBY2CQ0Kocuhm5UcPmAsFn_wytSzd5gYCXKE1yVV9nOE61mM9TrwUKDEP6IiAAIUm2TrXdKGOpbdk9HSwMawXRoy3fh_G9R2RbRjLAzgzed2fC3mLixflIFBepTDgsLpiIPNPAkxvhrC2gtsONLEH8Ij45Cgf0KvVZv54Uk2F4lB2kxyapqocTwXLA47agF7acLiI6yIlhW1_bmVfVZ4ljOIzVp44t_VgfT7FzV2SpBCczcamT-GX7SVDBQMNn0zAN6h4YQxPWYPPYeIcbrPb9kctubqdLvaIBuaRozEdMmHFQNrmmtCMukQJvcxWD7zszT6SQvLmAnvJ4wvfGVjoSmJ46Chq4z1s0a5zR5elJfEjVX-F5P80bBw7a2DRiSyDTIhy0tzu__BsecAA](https://microsoft.github.io/monaco-editor/playground.html?source=v0.52.2#XQAAAAIOCQAAAAAAAABBqQkHQ5NjdMjwa-jY7SIQ9S7DNlzs5W-mwj0fe1ZCDRFc9ws9XQE0SJE1jc2VKxhaLFIw9vEWSxW3yscxFLaASAEiAB2SUsdIicxj17f7KrRxrGCNFqb9_YD9EkAM-YhpywOMGyyECFSfTp8uXYWZSo9U-Lvsab7vckxbKJreAs-0Wrvz2LUbp1-rKe8ixMk-DZxSj47jZm1XcnjAmGurElvM4d9No9p-Hf70oJebGNsXqFWvkEP5Khi0secIsab2E8ydCQSLU-bRy8LXbhhmr0QcOoi_AoSXHuhld5AByp8NbvkoczlJ6EBxQTLsoXuns4Mq0eEFUFxPHDVk1Fgl2sMazUdBLvnoWUU5waOysMpbq4sKySkQsND3XOBoplChjTUzHGIcAEKcu5pwVuzDtJIfBUA8hJPxzrpgliTRlXEp3_f9O7hVIBszmaCWVbAYEGgK7TtWCHadS383K-wwxOb7fu4j85aFxtCco41R590_WScFmEnHL3pCzJGzmFfr-vCRY0N4PQgaoVhB4EGNUr7of7EQur-VDqn6M-UUX7CW9WzNcohLTLhWY03MqBUb1V6HrMJ1BkxiGSI9wkC-uwfhqQZqtC9pBZfi1okS01weR-betaUvYV1Mv8W6k__NzCu7mDoG5Xq8b8ixD8VANBfxas6BD_9krL7Mepkg_Ng6ZV0r2QdOkadlucCDd7fbhSqTw4wm_R0nb0wwITSl2dBFrxy-m2QEepGwgw8mlTDSquqZyt0z6BfFyO8eYnjH_X24Xyv36qzWTRxgMtREssP0nfgsxM3mEbKyMTYs0uiYoEvOrk2ydbeIOflTF3Id1fOxqYV7bPj5gKWIsLnokyJs8dNBC89DmdL-cItyj68jGxacGTyykyOBOiQO5B8swxFTA3kPz9aOWNT4giyEra8oqrzsGzexHJQFyxbP4D575d_GNVIV3AIUxUY59lTkPWVdP_7rGj0ZGyaXR-_L9-sZLFdxDeM9ZQGaFQMXr0N56BvtPI9cBi9yXsF52QAoZETTaVPmv1UggHDcaVqr_Hxp0dVODLj1kDM6MMN-mr3JsWNgyWalKq3aI_PRGfgd-PD4sGgsUtXjo7jS4eNb4yK7udwF1ylYuz-6lcS8Rt391O6DknntpxCm-bAtTW5Q8HiWgJEm2RAuekMZKovNUAbmhP_PEhh4xyJ5UCfI1vh-2Ga4NYdirmcDnGIdHph5f75CwFNXDQEWMf_0-6aN)).

**Reply by MolikoDeveloper** [reply]
# update
it does not include json validation.
![image](https://github.com/user-attachments/assets/04259868-c61d-4232-8651-abdbfeba8346)

**Reply by MolikoDeveloper** [reply]
with a basic linter [playground](https://microsoft.github.io/monaco-editor/playground.html?source=v0.52.2#XQAAAALQEgAAAAAAAABBqQkHQ5NjdMjwa-jY7SIQ9S7DNlzs5W-mwj0fe1ZCDRFc9ws9XQE0SJE1jc2VKxhaLFIw9vEWSxW3yscxcr710ohO0fTiA-VQbrkQ39DitycZlDIfGkpmkU3ubwsGDHBkr3JKW3BPqKPwzaLXPkO0E9Rne1mg_Dzoz4QAU1QRceMN6UR33jWLV6Cos9AWsezS9wccDHHoK2uO41XoMs1ev9nOiVCL3PY_iEieUC0HmfLbJyAMhyfcjA3RBjVuEPUjxtgxPr0hBQZQZNXe2BPRZAeFTX52aiUGGx0fgsYstextjLXSDqAoHs7-KDFrtI3LWo9wkkU85buEub1w3gqv5cJFYlDIY51qsH8scOZzWZs5xAj_yxRcq8JJKdwnmjVXH7ZXIOA2Ne1zl7qPhqkXoHyG2ljaMzi6IaxZeykInGLnakxKaESycZgD4E1canB14IPttOVNQoRD-IVYP81uEu3G3uVep3NaAgim7zqHc_ImtCzp8_z2wyryEk8fWxJ6-LySOgXG4CK9J3fzOt4XRz9EFS-W3mywR0PTOcoqEXPIlKwLkl1yGddqhiGtB4XS6t6XR2I83cVYx6HQVjIoGXU3m9omyM5XQXlE0cPVsqSA2BTwaCCfbDlRv-OaWkMfmHxFT6dKtbFeYkPSWFXmkf9AhfcIhstFUpbxQZMS6LAXbCC-5bxSXy6qC89eN3pYpZXrg52P7EeGmm0lrLDLb6FDzGX3bqZbJyb0iQGwnb7ekDySzS0CDlGoHf8KBK6VwonHBHKPVT7pFPrBaawb-mZwoZ1sKbc2iXM_a08KrNzprFGXp88h7vvN4JBAokTT8qsbeAJED4XqrcTwb8HGztMSTiLZscF5qR7ZD1ICZbM8kL5hvLKV2TiA0Qw31PzFPMpB9PqHPKlV6anMiWH8Go92b_SVCozL36gC_TjOF3lgjop4yulkQ02gsW-R2JZgwrYwhpJ59yG7GVrFKQPX4HuupJ0egs3mPQ-WjJ7iTEB_7dmFG6zpad2Vm5huyn8m-7YTFphhW0PiwDk3WOdqhttsqY64oMMGm4rsxAH6X8W10dfvqXRMzQzm2bbtDXnmFopexv384tSqFD3r1UyEOvzFJJGB8CdkFjBkZ97oTA7sW5mcU_AelmdaRfdnx851C6gR9nF8D2FHc4P55smX2DtAadaNGmgCxrQJZMAltJXDwUj8Cak35WZs3lUZt8j-hp1WvmWx6QD2vDScl7VO7cy4k9bVCRKVzawOJ9-FCKDpY2vca1miwz1YaOVP5LgmTU6q8mX5rCbC7S7zEOPnODEHxApFMv298OZQQhYlkvkWgU1a5bg3-3tqMIhHSnI56nHb8Ac1oFCDBlXmRqrcOqXJw20XEML7E8pg7tAov5QPe00fZqOJURRoDfHPcbgd-3p_LyzUBLghkv4hqJcgXPUKv6gc6dPq-0Ct_L4_2xgrjYWta6TN4c5PerIR9aMt61wy1wOPw_y5xorXUsnmJ9H2g8rPxyrRsW_9vZntfanmngn-kqimoBVjZsG5Ew7NvtcXV-ZndgtQoiIUBgDH8qh31G_0piCDEo6xBmraAPbgVsQrFx3g3ft4ijKoSDVrGk9IojhnxNwFNz4m5egphYZAAM4aafOm852RaUPOOrwKp23ncGRHjuQIWFrNNFQEVCLtKhuGQup_Cr5Kqd_HA9Q1XTbGVe6NSK2B0rtO2rXuPXk9yPzOA8aunc1SzSa3YiB3rC_T_UgUhe-kdfViztdKwXN_MMDM2_Q243gDT1-ajvCtXdG0A-yRT64Ymlgfd6CWkgMofc68b_uIJSX60bt3amIUU4hmppsxmImSQlcfcZqntIvOOHRmiMfhJrQdDakl-Uv11J-l3HAhGCn40WJ14i4UDs0Y6P271Kie7rkAd62t3BQeD0jAYRWu_RR436XZa4sV_ipEypnWOTa8aLsdqQ4enN4BWsE5HBCJPAwYdx19gtLe9N6S27kU44djlCQgv2LzXX4ERlqdRhSlwHO47t2pEALi0205wuTuuMFrXjr3thz1gxFnQ6bIpPXgtkH61BysbZAuY9eoEGyIXqNFvMoXRY5Z7WFZQB1rugJn6WigYxwly4W3MevW6wxPivu7_GtT6ukp9sUw9nZJc25Qxt4ND_TT3btW7x8zSXZPnasnJHUkj2qlICEEJLKB4PjVEaisaLqYoKtl89K4glXYNDAsxZ2ON-x51AReCjc6k9OoPdAozfJqwFyIYOeLBsHq3IypxHZ6s1dFN28pI7Ly7TK1Jdd5t9P__VaIYw)
![image](https://github.com/user-attachments/assets/80bd5df2-67ee-460c-be0c-ac844fcb6a05)

**Reply by MolikoDeveloper** [reply]
i hope it helps.
